scispacy/linking.py

```diff
@@ -96,20 +96,26 @@ def __init__(
         self.umls = self.kb
 
     def __call__(self, doc: Doc) -> Doc:
-        mentions = []
+        mention_strings = []
         if self.resolve_abbreviations and Doc.has_extension("abbreviations"):
-
+            # This used to create a mentions list of spans, just to create a list of mention strings later on
             for ent in doc.ents:
-                # TODO: This is possibly sub-optimal - we might
```

Comment:
I don't think this is quite this straightforward. The later part (line 115 in original) assumes that `mention_strings` is the same length as `doc.ents`. I guess we're not testing the combination of the abbreviation detector and the linker very well...

I would just do the Span vs str check for now. But also if you could fix the test so your proposed code fails it (or explain why I'm thinking about it wrong), that would be great.

Reply:
I'll make this simpler tomorrow and then start working on the more complex version when I have some more time